module/database/UserDatabase.py

```diff
@@ -42,8 +43,8 @@ def checkAuth(db, user, password):
 
     @style.queue
     def addUser(db, user, password):
-        salt = reduce(lambda x, y: x + y, [str(random.randint(0, 9)) for i in range(0, 5)])
-        h = sha1(salt + password)
+        salt = functools.reduce(lambda x, y: x + y, [str(random.randint(0, 9)) for i in range(0, 5)])
+        h = sha1(salt.encode('utf-8') + password.encode('utf-8'))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of insecure MD2, MD4, MD5, or SHA1 hash function.](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=4846282)